projects/spectator/test/defer-block.spec.ts

```diff
@@ -0,0 +1,222 @@
+import { Component } from '@angular/core';
+import { DeferBlockBehavior, fakeAsync } from '@angular/core/testing';
+import { createComponentFactory } from '@ngneat/spectator';
+
+describe('DeferBlock', () => {
+  describe('Playthrough Behavior', () => {
+    @Component({
+      selector: 'app-root',
+      template: `
+        <button data-test="button--isVisible" (click)="isVisible = !isVisible">Toggle</button>
+
+        @defer (when isVisible) {
+        <div>empty defer block</div>
+        } ,
+      `,
+      standalone: true,
+    })
+    class DummyComponent {
+      isVisible = false;
+    }
+
+    const createComponent = createComponentFactory({
+      component: DummyComponent,
+      deferBlockBehavior: DeferBlockBehavior.Playthrough,
```

Comment:
What do you say about also support here the state?

```ts
 const createComponent = createComponentFactory({
      component: DummyComponent,
      deferBlockBehavior: DeferBlockBehavior.Playthrough,
      deferBlockState: state,
      deferBlockIndex: optionalIndex
})
```

Reply:
I think we can do that here, https://github.com/profanis/spectator/blob/3f63c68440f2dc11cb251e49d8ebdc8a9a44a5e9/projects/spectator/src/lib/spectator/create-factory.ts#L162

```ts
    // e.g.
    spectator.deferBlock(1).renderComplete()

    return spectator;
```

Reply:
Hmm, I need to think of an elegant way for this. The `render*` methods return a promise and it won't work nicely in the `create-factory`

Reply:
Ok, we can add it in in a different PR.